eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -370,55 +371,59 @@ object Helpers {
-        Transactions.addSigs(claimDelayed, sig)
-      })
+      val mainDelayedTx = makeDelayedTx(tx, "main-delayed-output")
 
       // those are the preimages to existing received htlcs
       val preimages = commitments.localChanges.all.collect { case u: UpdateFulfillHtlc => u.paymentPreimage }
+      val htlcAmounts = commitments.localCommit.spec.htlcs.map(htlc => htlc.add.paymentHash -> htlc.add.amountMsat).toMap
 
-      val htlcTxes = localCommit.publishableTxs.htlcTxsAndSigs.collect {
-        // incoming htlc for which we have the preimage: we spend it directly
-        case HtlcTxAndSigs(txinfo@HtlcSuccessTx(_, _, paymentHash), localSig, remoteSig) if preimages.exists(r => sha256(r) == paymentHash) =>
-          generateTx("htlc-success")(Try {
-            val preimage = preimages.find(r => sha256(r) == paymentHash).get
-            Transactions.addSigs(txinfo, localSig, remoteSig, preimage)
-          })
+      val successAndDelayedTxs = for {
+        HtlcTxAndSigs(txinfo@HtlcSuccessTx(_, _, paymentHash), localSig, remoteSig) <- localCommit.publishableTxs.htlcTxsAndSigs
+        if preimages.exists(r => sha256(r) == paymentHash)
 
-        // (incoming htlc for which we don't have the preimage: nothing to do, it will timeout eventually and they will get their funds back)
+        success <- generateTx("htlc-success")(Try {
+          val preimage = preimages.find(r => sha256(r) == paymentHash).get
+          Transactions.addSigs(txinfo, localSig, remoteSig, preimage)
+        })
 
-        // outgoing htlc: they may or may not have the preimage, the only thing to do is try to get back our funds after timeout
-        case HtlcTxAndSigs(txinfo: HtlcTimeoutTx, localSig, remoteSig) =>
-          generateTx("htlc-timeout")(Try {
-            Transactions.addSigs(txinfo, localSig, remoteSig)
-          })
-      }.flatten
-
-      // all htlc output to us are delayed, so we need to claim them as soon as the delay is over
-      val htlcDelayedTxes = htlcTxes.flatMap {
-        txinfo: TransactionWithInputInfo =>
-          generateTx("claim-htlc-delayed")(Try {
-            val claimDelayed = Transactions.makeClaimDelayedOutputTx(
-              txinfo.tx,
-              Satoshi(localParams.dustLimitSatoshis),
-              localRevocationPubkey,
-              remoteParams.toSelfDelay,
-              localDelayedPubkey,
-              localParams.defaultFinalScriptPubKey, feeratePerKwDelayed)
-            val sig = keyManager.sign(claimDelayed, keyManager.delayedPaymentPoint(localParams.channelKeyPath), localPerCommitmentPoint)
-            Transactions.addSigs(claimDelayed, sig)
-          })
-      }
+        delayed <- makeDelayedTx(txinfo.tx, "claim-htlc-success-delayed")
+        _ = eventStream.publish(PaymentSettlingOnChain(offChainAmount = MilliSatoshi(htlcAmounts(paymentHash)),
+          onChainAmount = delayed.tx.txOut.head.amount, paymentHash, delayed.tx.txid, "claim-htlc-delayed", isDone = false))
```

Comment:
`def claimCurrentLocalCommitTxOutputs` internal logic was changed to allow easy access to all parameters needed in `PaymentSettlingOnChain` message. 

A small additional benefit, I think, is that `success.tx -> delayed.tx` will only be produced if both transactions can make it to chain whereas previously `success` was always being published, even if related `delayed` did not have enough fees. This removes a useless transaction from blockchain.